## Построение модели случайного леса

In [3]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report
from utils.preformatting_data import preformat
import pandas as pd

# Загрузка данных
data = preformat(pd.read_csv('../medics_1.csv', delimiter=',', encoding='utf-8'))

In [6]:
# Определение признаков и целевой переменной
X = data[["возраст", "время_пережатия_аорты", "объем_кровопотерии",
           "есть_хроническое_заболевание", "переливание_крови_и_аик",
           "чсс", "мочевина", "длительность_аик"]]
y = data["развитие_опп"]

# Разделение данных на обучающий и тестовый наборы
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Инициализация модели Случайного леса
random_forest = RandomForestClassifier(n_estimators=100, random_state=42)

# Обучение модели
random_forest.fit(X_train, y_train)

RandomForestClassifier(random_state=42)

In [8]:
# Прогнозы на тестовом наборе
y_pred = random_forest.predict(X_test)

In [9]:
# Оценка модели
accuracy = accuracy_score(y_test, y_pred)
print("Accuracy:", accuracy)
print("Classification Report:")
print(classification_report(y_test, y_pred))

Accuracy: 0.6875
Classification Report:
              precision    recall  f1-score   support

           0       0.38      0.38      0.38         8
           1       0.79      0.79      0.79        24

    accuracy                           0.69        32
   macro avg       0.58      0.58      0.58        32
weighted avg       0.69      0.69      0.69        32



### Оценка модели Случайного леса для классификации "Развитие ОПП"

- **Точность (Accuracy)**: 68.75%
- **Precision (точность)**:
  - Класс "0" (Отсутствие развития ОПП): 38%
  - Класс "1" (Наличие развития ОПП): 79%
- **Recall (полнота)**:
  - Класс "0": 38%
  - Класс "1": 79%
- **F1-score (F-мера)**:
  - Класс "0": 38%
  - Класс "1": 79%
- **Поддержка (Support)**: 8 для класса "0", 24 для класса "1"

#### Общий вывод:

Модель Случайного леса успешно предсказывает наличие или отсутствие развития ОПП с точностью 68.75%. Она достаточно хорошо предсказывает класс наличие развития ОПП, но менее эффективна в предсказании класса отсутствия развития ОПП. Значение F1-меры составляет 58%, что указывает на среднюю эффективность модели в целом. В сравнении с моделью с градиентным бустингом, эта модель показала себя хуже.
